src/main/java/com/minecolonies/coremod/colony/buildings/BuildingHome.java

```diff
@@ -238,6 +315,15 @@ public boolean hasResident(final CitizenData citizen)
     }
 
     /**
+     * Checks if food in the home is required.
+     * If yes set foodNeeded to true, else to false.
+     */
+    public void checkIfFoodNeeded()
+    {
+        setFoodNeeded(residents.stream().filter(resident -> resident.getSaturation() < EntityCitizen.HIGH_SATURATION).findFirst().isPresent());
+    }
+
+    /**
      * The view of the citizen hut.
      */
     public static class View extends AbstractBuildingHut.View
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this class from 38 to the maximum allowed 30 or externalize it in a public class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)